thanos/charts/templates/query/ingress-grpc.yaml

```
@@ -0,0 +1,40 @@
+{{- if .Values.thanos.query.ingress.grpc.enabled -}}
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  labels:
+    {{- include "plugin.labels" . | nindent 4 }}
+    {{- include "thanos.labels" . | nindent 4 }}
+  {{- with .Values.thanos.query.ingress.grpc.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  name: {{ include "release.name" . }}-grpc
+spec:
+  ingressClassName: nginx
+  rules:
+    {{- range .Values.thanos.query.ingress.grpc.hosts }}
+    - host: {{ .host | quote }}
+      http:
+        paths:
+          {{- range .paths }}
+          - path: {{ .path }}
+            pathType: {{ .pathType }}
+            backend:
+              service:
+                name: {{ include "release.name" . }}-query
```

This will not work. In the scope of the range you can work with `$` to include templates.
```suggestion
                name: {{ include "release.name" $ }}-query
```